providers/os/id/platform.go

```diff
@@ -161,6 +162,17 @@ func GatherPlatformInfo(conn shared.Connection, pf *inventory.Platform, idDetect
 			}, hostErr
 		}
 		return &PlatformInfo{}, nil
+	case idDetector == ids.IdDetector_SerialNumber:
+		serial, err := serialnumber.SerialNumber(conn, pf)
```

Comment:
Have we tested that for macos, windows and linux?

Reply:
This is taken from the existing resource - `machine.system.serial` - which works on all aforementioned OS's
My tests so far show that the on the systems which don't provide that data (for example CI containers) serial evaluates to empty string, so it's omitted in the `asset.ids`